src/rngs/os.rs

```diff
@@ -1102,6 +1102,76 @@ mod imp {
     }
 }
 
+#[cfg(all(target_arch = "wasm32",
+          not(target_os = "emscripten"),
+          not(feature = "stdweb"),
+          feature = "wasm-bindgen"))]
+mod imp {
+    use core::fmt;
+
+    use __wbg_shims::*;
+
+    use {Error, ErrorKind};
+    use super::OsRngImpl;
+
+    #[derive(Clone)]
+    pub enum OsRng {
```

Comment:
Is this definition exposed publicly, or does it get wrapped? If the former, might want to hide the fact that it is an enum:

```rust
pub struct OsRng(OsRngKind);

enum OsRngKind {
    Node(..),
    Browser(..),
}
```

Reply:
Ah yeah this is [wrapped above](https://github.com/rust-lang-nursery/rand/blob/0faff206d21515e3fd5233cfbd6899efd673a46b/src/rngs/os.rs#L118)